src/prototype_creep.js

```diff
@@ -102,8 +102,7 @@ Creep.prototype.checkForHandle = function() {
   const role = this.memory.role;
   if (!role) {
     this.log('Creep role not defined for: ' + this.id + ' ' + this.name.split('-')[0].replace(/[0-9]/g, ''));
-    this.memory.killed = true;
-    this.suicide();
+    Creep.recycleCreep(this);
```

Comment:
i dont know if this is working... did you test it? 
if creep.memory.role is not set can a creep then move to a spawn for recycle?

Reply:
Good question.
Before, I use the new added implemented function, so I know it can, but now it is an existed function...
I will give it a role here, so it would think itself be a scout.